File: dl-invites/app/src/main/java/com/google/firebase/dynamicinvites/view/AdvancedActivity.java
package com.google.firebase.dynamicinvites.view;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;

import com.google.firebase.dynamicinvites.R;
import com.google.firebase.dynamicinvites.presenter.InvitePresenter;

public class AdvancedActivity extends AppCompatActivity implements ShareDialogFragment.Listener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_advanced);

        findViewById(R.id.buttonShare).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ShareDialogFragment.newInstance().show(getSupportFragmentManager(), "dialog");
            }
        });
    }

    @Override
    public void onItemClicked(InvitePresenter presenter) {
        presenter.sendInvite(this);
    }
}

File: dl-invites/app/src/main/java/com/google/firebase/dynamicinvites/kotlin/view/AdvancedActivity.kt
package com.google.firebase.dynamicinvites.kotlin.view

import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

import com.google.firebase.dynamicinvites.R
import com.google.firebase.dynamicinvites.kotlin.presenter.InvitePresenter

class AdvancedActivity : AppCompatActivity(), ShareDialogFragment.Listener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_advanced)

        findViewById<Button>(R.id.buttonShare).setOnClickListener {
            ShareDialogFragment.newInstance().show(supportFragmentManager, "dialog")
        }
    }

    override fun onItemClicked(presenter: InvitePresenter) {
        presenter.sendInvite(this)
    }
}
